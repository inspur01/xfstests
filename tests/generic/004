#! /bin/bash
# FS QA Test No. 004
#
# Test O_TMPFILE opens, and linking them back into the namespace.
#
#-----------------------------------------------------------------------
# Copyright (c) 2014 Christoph Hellwig.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
    cd /
    rm -f ${testfile}
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

# real QA test starts here
_supported_fs generic
_supported_os Linux

_require_test
_require_xfs_io_command "flink"

rm -f $seqres.full

testfile="${TEST_DIR}/tst-tmpfile-flink"

# test creating a r/w tmpfile, do I/O and link it into the namespace
$XFS_IO_PROG -T \
	-c "pwrite 0 4096" \
	-c "pread 0 4096" \
	-c "flink ${testfile}" \
	${TEST_DIR} | _filter_xfs_io

rm ${testfile}

# test creating a r/o tmpfile.  Should fail
$XFS_IO_PROG -Tr ${TEST_DIR} -c "close" 2>&1 | _filter_test_dir

# success, all done
status=0
exit
