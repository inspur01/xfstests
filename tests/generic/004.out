QA output created by 004
wrote 4096/4096 bytes at offset 0
XXX Bytes, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)
read 4096/4096 bytes at offset 0
XXX Bytes, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)
TEST_DIR: Invalid argument
